dwave/plugins/torch/models/boltzmann_machine.py

```diff
@@ -662,3 +662,277 @@ def estimate_beta(self, spins: torch.Tensor) -> float:
+            weight_decay (float): weight decay (L2 regularization) coefficient for weights.
+            n_epochs (int): Number of training epochs.
+
+        Returns:
+            torch.Tensor: The reconstruction error (L1 norm) for the batch.
+
+        """
+
+        # Positive phase (data-driven)
+        hidden_probs = torch.sigmoid(self._hidden_biases + batch @ self._weights)
+
+        weight_grads = torch.matmul(batch.t(), hidden_probs)
+        visible_bias_grads = batch
+        hidden_bias_grads = hidden_probs
+
+        batch_size = batch.size(0)
+        
+        # Initialize previous visible states for Persistent CD
+        if self._previous_visible_values == None:
+            self._previous_visible_values = torch.randn_like(
+                batch, device=self._weights.device
+            )
+
+        # Negative phase (model-driven)
+        # Sample from the model using gibbs sampling
+        visible_values, hidden_values = self.generate_sample(
+            batch_size, n_gibbs_steps, self._previous_visible_values
+        )
+
+        visible_values = visible_values.detach()
+        hidden_values = hidden_values.detach()
+        # Store samples to initialize the next Markov chain with (PCD)
+        self._previous_visible_values = visible_values
+
+        # Compute the gradients for negative phase
+        weight_grads -= torch.matmul(visible_values.t(), hidden_values)
+
+        visible_bias_grads -= visible_values
+        hidden_bias_grads -= hidden_values
+
+        # Average across the batch
+        weight_grads /= batch_size
+        visible_bias_grads /= batch_size
+        hidden_bias_grads /= batch_size
+
+        # Compute decayed learning rate
+        decayed_learning_rate = learning_rate - (learning_rate / n_epochs * epoch)
+
+        # Update momenta
+        self._weight_momenta = self._weight_momenta * momentum_coefficient + decayed_learning_rate * weight_grads
+        self._visible_bias_momenta = self._visible_bias_momenta * momentum_coefficient + decayed_learning_rate * torch.sum(
+            visible_bias_grads, dim=0
+        )
+        self._hidden_bias_momenta = self._hidden_bias_momenta * momentum_coefficient + decayed_learning_rate * torch.sum(
+            hidden_bias_grads, dim=0
+        )
+
+        with torch.no_grad():
+            # Update parameters
+            self._weights += self._weight_momenta
+            self._visible_biases += self._visible_bias_momenta
+            self._hidden_biases += self._hidden_bias_momenta
+
+            # Apply weight decay
+            self._weights -= decayed_learning_rate * self._weights * weight_decay
+
+        # Compute reconstruction error (L1 norm)
+        reconstruction = self._sample_visible(self._sample_hidden(batch))
+        reconstruction = reconstruction.detach()
+        error = torch.sum(torch.abs(batch - reconstruction))
+
+        return error
+
+    def forward(self, visible: torch.Tensor) -> torch.Tensor:
```

Comment:
To keep the same as the GRBM.
```suggestion
    def forward(self, x: torch.Tensor) -> torch.Tensor:
```